libqtile/widget/base.py

```diff
@@ -301,17 +302,6 @@ def __init__(self, text=" ", width=bar.CALCULATED, **config):
         self.text = text
         self.add_defaults(_TextBox.defaults)
 
-    @property
-    def text(self):
-        return self._text
-
-    @text.setter
-    def text(self, value):
-        assert value is None or isinstance(value, str)
-        self._text = value
-        if self.layout:
-            self.layout.text = value
-
```

Comment:
I decided to remove this whole part because it felt useless to me to 1/ re-assign `self.layout.text` every time you manipulate `self.text`, even if you only use it when it's drawn and 2/ check the type of what's given in with an `assert` (let the layout fail itself if it can't handle this)

Reply:
This decision ended up being a bit buggy, see https://github.com/qtile/qtile/commit/77b47b07b08bdd38e15468714f3ba2d5de56a78e